cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2630,46 +2628,101 @@ macro(build_awssdk)
   endif()
 
   set(AWSSDK_CMAKE_ARGS
-      -DCMAKE_BUILD_TYPE=Release
+      -DCMAKE_BUILD_TYPE=release
+      -DMINIMIZE_SIZE=on
+      -DBUILD_ONLY=config\\$<SEMICOLON>s3\\$<SEMICOLON>transfer\\$<SEMICOLON>identity-management\\$<SEMICOLON>sts
       -DCMAKE_INSTALL_LIBDIR=lib
-      -DBUILD_ONLY=s3;core;config;identity-management;sts
       -DENABLE_UNITY_BUILD=on
+      -DBUILD_SHARED_LIBS=off
       -DENABLE_TESTING=off
       "-DCMAKE_C_FLAGS=${EP_C_FLAGS}"
       "-DCMAKE_INSTALL_PREFIX=${AWSSDK_PREFIX}")
 
-  set(
-    AWSSDK_CORE_SHARED_LIB
-    "${AWSSDK_PREFIX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}aws-cpp-sdk-core${CMAKE_SHARED_LIBRARY_SUFFIX}"
-    )
-  set(
-    AWSSDK_S3_SHARED_LIB
-    "${AWSSDK_PREFIX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}aws-cpp-sdk-s3${CMAKE_SHARED_LIBRARY_SUFFIX}"
-    )
-  set(
-    AWSSDK_IAM_SHARED_LIB
-    "${AWSSDK_PREFIX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}aws-cpp-sdk-identity-management${CMAKE_SHARED_LIBRARY_SUFFIX}"
-    )
-  set(
-    AWSSDK_STS_SHARED_LIB
-    "${AWSSDK_PREFIX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}aws-cpp-sdk-sts${CMAKE_SHARED_LIBRARY_SUFFIX}"
-    )
-  set(AWSSDK_SHARED_LIBS "${AWSSDK_CORE_SHARED_LIB}" "${AWSSDK_S3_SHARED_LIB}"
-                         "${AWSSDK_IAM_SHARED_LIB}" "${AWSSDK_STS_SHARED_LIB}")
+  set(AWSSDK_COMMON_STATIC_LIB "${AWSSDK_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}aws-c-common${CMAKE_STATIC_LIBRARY_SUFFIX}")
```

Comment:
Sadly we can't use the FindAWSSDK cmake files to locate the SDK since they are available after the external projects gets built.